test/modules/TestResultCoordinator/Reports/DirectMethod/DirectMethodReportMetadata.cs

```diff
@@ -22,6 +22,8 @@ public DirectMethodReportMetadata(string senderSource, TimeSpan tolerancePeriod,
 
         public Option<string> ReceiverSource { get; }
 
+        public string NetworkControllerRunProfileName => Settings.Current.NetworkControllerRunProfileName;
```

Comment:
create a field and pass in from constructor.

Reply:
Talked offline. Instead we are going to pass it into ReportGeneratorFactory